server/src/main/java/org/eclipse/lsp/cobol/core/engine/ThreadInterruptionUtil.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.engine;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Utility class to check if a current running thread is interrupted. Throws {@link
+ * UserInterruptException} in case thread is interrupted.
+ */
+@UtilityClass
+@Slf4j
+public class ThreadInterruptionUtil {
+
+  /**
+   * Utility method to check if a current running thread is interrupted.
+   *
+   * @throws UserInterruptException in case thread is interrupted.
+   * @param methodName invoking method name.
+   */
+  public void checkThreadInterrupted(String methodName) {
+    if (Thread.interrupted()) {
+      LOG.debug(methodName + " is interrupted by user");
+      throw new UserInterruptException("Parsing interrupted by user.");
```

Comment:
I'm just thinking. The extension will be logged with its call stack, so we will see the caller name in the log in any case. Maybe, we don't need to pass it to this method?